cmd/eval-dev-quality/cmd/report.go

```diff
@@ -0,0 +1,109 @@
+package cmd
+
+import (
+	"bufio"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	pkgerrors "github.com/pkg/errors"
+
+	"github.com/symflower/eval-dev-quality/evaluate/report"
+	"github.com/symflower/eval-dev-quality/log"
+)
+
+// Report holds the "report" command.
+type Report struct {
+	// EvaluationPaths holds the list of file paths where the results of previous evaluations are stored.
+	EvaluationPaths []string `long:"evaluation-path" description:"File path for an evaluation CSV file where the results of a previous evaluation are stored."`
+	// ResultPath holds the directory path where the overall results should be written to.
+	ResultPath string `long:"result-path" description:"Directory path were the overall results are written to."`
+
+	// logger holds the logger of the command.
+	logger *log.Logger
+}
+
+var _ SetLogger = (*Evaluate)(nil)
+
+// SetLogger sets the logger of the command.
+func (command *Report) SetLogger(logger *log.Logger) {
+	command.logger = logger
+}
+
+// Execute executes the command.
+func (command *Report) Execute(args []string) (err error) {
+	evaluationCSVFilePath, err := os.OpenFile(filepath.Join(command.ResultPath, "evaluation.csv"), os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0755)
+	if err != nil {
+		command.logger.Panicf("ERROR: %s", err)
+	}
+	defer evaluationCSVFilePath.Close()
+
+	evaluationCSVWriter := bufio.NewWriter(evaluationCSVFilePath)
```

Comment:
Would move the Writer below the gathering of records, and check if there are any records to begin with.